### PR TITLE
Improve CI script to fail earlier 

### LINKED ISSUE #548: CI scripts should bail at first sign of failure
Some ci scripts continue to run after a clear failure:
https://travis-ci.org/bluerobotics/ping-viewer/jobs/559768707#L907

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #548

### CODE REVIEW

**Comment by Williangalvani** [comment]
It looks like the CI stalled

**Comment by patrickelectric** [comment]
> It looks like the CI stalled

this is a common thing in travis.

File: tools/runtests.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ rm -rf $build_test
 mkdir -p ${build_test}
 qmake -o ${build_test} -r -Wall -Wlogic -Wparser CONFIG+=test ${projectpath}
 make -C ${build_test}
-$build_test/pingviewer
+xvfb-run --server-args="-screen 0 1024x768x24" $build_test/pingviewer || exit 1
```
Comment (Williangalvani) [inline tools/runtests.sh]: what exactly does it change?

Comment (patrickelectric) [reply tools/runtests.sh]: it moves the test to run in a fake display, it appears that it was failing all this time.

**Pull request merged**